config.go

```diff
@@ -443,6 +443,47 @@ func (c *Config) GetAll(alias, key string) ([]string, error) {
 	return all, nil
 }
 
+// HostsAliases returns all hosts aliases in the configuration that match the
+// given pattern, or nil if none are present.
+// A host alias matches when its first pattern matches and has a KV node that
+// contains key 'Hostname'.
+func (c *Config) HostsAliases(pattern string) ([]string, error) {
+	p, err := NewPattern(pattern)
+	if err != nil {
+		return nil, err
+	}
+	all := []string(nil)
+	for _, host := range c.Hosts {
+		alias := host.Patterns[0].String()
+		ok := alias == "*" || p.regex.MatchString(alias)
+		if !ok || p.not {
+			continue
+		}
+
+		for _, node := range host.Nodes {
+			switch t := node.(type) {
+			case *Empty:
+				continue
+			case *KV:
+				// "keys are case-insensitive" per the spec
```

Comment:
We should probably use strings.EqualFold here to match t.Key and "hostname"

Reply:
done